megatron/core/extensions/transformer_engine.py

```diff
@@ -49,6 +51,56 @@ def condition_init_method(config, init_method):
     """Condition TE init_method on config.perform_initialization."""
     return init_method if config.perform_initialization else (lambda w: None)
 
+class TEActivationOp:
+    """
+    A conditional wrapper to initialize an instance of Transformer-Engine's activation
+    function operators (e.g. Silu, SwiGLU, etc)
+    """
+    def __new__(cls, config: TransformerConfig):
+        instance = None
+        if config.gated_linear_unit:
+            if config.activation_func == F.silu:
+                instance = te_ops.SwiGLU()
+            elif config.activation_func == F.gelu:
+                instance = te_ops.GEGLU()
+            elif config.activation_func == F.silu:
+                instance = te_ops.ReGLU()
+        else:
+            if config.activation_func == F.gelu:
+                instance = te_ops.GELU()
+            elif config.activation_func == F.silu:
+                instance = te_ops.ReLU()
+        if instance is None:
+            raise Exception('Only SwiGLU, GEGLU, ReGLU, GELU, ReLU are supported by '
+            'transformer engine. Consider setting use_te_activation_func=False')
+        return te_ops.Sequential(instance)
+
+class TEActivationOpFp8:
```

Comment:
Why do we need this?
In case of fp8 recipe, the fp8 quantization should be handled by `linear module`, right?

Also, `te_ops.Quantize` doesn't cast using scale_factor from amax history, right?